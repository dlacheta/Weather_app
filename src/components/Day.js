import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

const Day = ({ today, weather }) => (
  <div className="card">
    <div className="card-body">
      <h5 className="card-title">
        {today
          ? `Today (${moment(weather[0].dt_txt).format('DD - MMM - YYYY')})`
          : `${moment(weather[0].dt_txt).format('dddd')} (${moment(weather[0].dt_txt).format(
            'DD - MMM - YYYY',
          )})`}
      </h5>
      <ul className="list-group">
        {weather.map(time => (
          <li className="d-flex list-group-item align-middle" key={time.dt}>
            <div className="time">{moment(time.dt_txt).format('HH:mm')}</div>
            <div className="icon">
              <img
                src={`http://openweathermap.org/img/w/${time.weather[0].icon}.png`}
                alt={time.weather[0].description}
              />
            </div>
            <div className="temp">
              {time.main.temp}
              <sup>o</sup>
C
            </div>
            <div className="hpa">{`${time.main.pressure} hPa`}</div>
            <div className="wind">{time.wind.speed}</div>
            <div className="wind-arrow">
              <i className="material-icons" style={{ transform: `rotate(${time.wind.deg}deg)` }}>
                arrow_right_alt
              </i>
            </div>
          </li>
        ))}
      </ul>
    </div>
  </div>
);

Day.propTypes = {
  today: PropTypes.bool.isRequired,
  weather: PropTypes.arrayOf(
    PropTypes.shape({
      dt: PropTypes.number,
      dt_txt: PropTypes.string,
      weather: PropTypes.arrayOf(
        PropTypes.shape({
          icon: PropTypes.string,
          description: PropTypes.string,
        }),
      ),
      main: PropTypes.shape({
        temp: PropTypes.number,
        pressure: PropTypes.number,
      }),
      wind: PropTypes.shape({
        speed: PropTypes.number,
        deg: PropTypes.number,
      }),
    }),
  ).isRequired,
};

export default Day;
