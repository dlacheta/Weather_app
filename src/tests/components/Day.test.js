import React from 'react';
import { shallow } from 'enzyme';
import Day from '../../components/Day';

it('Day component renders correctly', () => {
  const today = true;
  const weather = [
    {
      dt: 1539248400,
      dt_txt: '2018-10-11 12:00:00',
      weather: [
        {
          icon: '01d',
          description: 'clear sky',
        },
      ],
      main: {
        temp: 286.164,
        pressure: 1017.58,
      },
      wind: {
        speed: 3.61,
        deg: 165.001,
      },
    },
  ];
  const wrapper = shallow(<Day today={today} weather={weather} />);
  expect(wrapper).toMatchSnapshot();
});
