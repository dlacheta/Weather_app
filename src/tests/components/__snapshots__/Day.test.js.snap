// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Day
    today={true}
    weather={
      Array [
        Object {
          "dt": 1539248400,
          "dt_txt": "2018-10-11 12:00:00",
          "main": Object {
            "pressure": 1017.58,
            "temp": 286.164,
          },
          "weather": Array [
            Object {
              "description": "clear sky",
              "icon": "01d",
            },
          ],
          "wind": Object {
            "deg": 165.001,
            "speed": 3.61,
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          Today (11 - Oct - 2018)
        </h5>
        <ul
          className="list-group"
        >
          <li
            className="d-flex list-group-item align-middle"
          >
            <div
              className="time"
            >
              12:00
            </div>
            <div
              className="icon"
            >
              <img
                alt="clear sky"
                src="http://openweathermap.org/img/w/01d.png"
              />
            </div>
            <div
              className="temp"
            >
              286.164
              <sup>
                o
              </sup>
              C
            </div>
            <div
              className="hpa"
            >
              1017.58 hPa
            </div>
            <div
              className="wind"
            >
              3.61
            </div>
            <div
              className="wind-arrow"
            >
              <i
                className="material-icons"
                style={
                  Object {
                    "transform": "rotate(165.001deg)",
                  }
                }
              >
                arrow_right_alt
              </i>
            </div>
          </li>
        </ul>
      </div>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5
            className="card-title"
          >
            Today (11 - Oct - 2018)
          </h5>,
          <ul
            className="list-group"
          >
            <li
              className="d-flex list-group-item align-middle"
            >
              <div
                className="time"
              >
                12:00
              </div>
              <div
                className="icon"
              >
                <img
                  alt="clear sky"
                  src="http://openweathermap.org/img/w/01d.png"
                />
              </div>
              <div
                className="temp"
              >
                286.164
                <sup>
                  o
                </sup>
                C
              </div>
              <div
                className="hpa"
              >
                1017.58 hPa
              </div>
              <div
                className="wind"
              >
                3.61
              </div>
              <div
                className="wind-arrow"
              >
                <i
                  className="material-icons"
                  style={
                    Object {
                      "transform": "rotate(165.001deg)",
                    }
                  }
                >
                  arrow_right_alt
                </i>
              </div>
            </li>
          </ul>,
        ],
        "className": "card-body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Today (11 - Oct - 2018)",
            "className": "card-title",
          },
          "ref": null,
          "rendered": "Today (11 - Oct - 2018)",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="d-flex list-group-item align-middle"
              >
                <div
                  className="time"
                >
                  12:00
                </div>
                <div
                  className="icon"
                >
                  <img
                    alt="clear sky"
                    src="http://openweathermap.org/img/w/01d.png"
                  />
                </div>
                <div
                  className="temp"
                >
                  286.164
                  <sup>
                    o
                  </sup>
                  C
                </div>
                <div
                  className="hpa"
                >
                  1017.58 hPa
                </div>
                <div
                  className="wind"
                >
                  3.61
                </div>
                <div
                  className="wind-arrow"
                >
                  <i
                    className="material-icons"
                    style={
                      Object {
                        "transform": "rotate(165.001deg)",
                      }
                    }
                  >
                    arrow_right_alt
                  </i>
                </div>
              </li>,
            ],
            "className": "list-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1539248400",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="time"
                  >
                    12:00
                  </div>,
                  <div
                    className="icon"
                  >
                    <img
                      alt="clear sky"
                      src="http://openweathermap.org/img/w/01d.png"
                    />
                  </div>,
                  <div
                    className="temp"
                  >
                    286.164
                    <sup>
                      o
                    </sup>
                    C
                  </div>,
                  <div
                    className="hpa"
                  >
                    1017.58 hPa
                  </div>,
                  <div
                    className="wind"
                  >
                    3.61
                  </div>,
                  <div
                    className="wind-arrow"
                  >
                    <i
                      className="material-icons"
                      style={
                        Object {
                          "transform": "rotate(165.001deg)",
                        }
                      }
                    >
                      arrow_right_alt
                    </i>
                  </div>,
                ],
                "className": "d-flex list-group-item align-middle",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "12:00",
                    "className": "time",
                  },
                  "ref": null,
                  "rendered": "12:00",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="clear sky"
                      src="http://openweathermap.org/img/w/01d.png"
                    />,
                    "className": "icon",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "clear sky",
                      "src": "http://openweathermap.org/img/w/01d.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      286.164,
                      <sup>
                        o
                      </sup>,
                      "C",
                    ],
                    "className": "temp",
                  },
                  "ref": null,
                  "rendered": Array [
                    286.164,
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "o",
                      },
                      "ref": null,
                      "rendered": "o",
                      "type": "sup",
                    },
                    "C",
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1017.58 hPa",
                    "className": "hpa",
                  },
                  "ref": null,
                  "rendered": "1017.58 hPa",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 3.61,
                    "className": "wind",
                  },
                  "ref": null,
                  "rendered": 3.61,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      className="material-icons"
                      style={
                        Object {
                          "transform": "rotate(165.001deg)",
                        }
                      }
                    >
                      arrow_right_alt
                    </i>,
                    "className": "wind-arrow",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "arrow_right_alt",
                      "className": "material-icons",
                      "style": Object {
                        "transform": "rotate(165.001deg)",
                      },
                    },
                    "ref": null,
                    "rendered": "arrow_right_alt",
                    "type": "i",
                  },
                  "type": "div",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            Today (11 - Oct - 2018)
          </h5>
          <ul
            className="list-group"
          >
            <li
              className="d-flex list-group-item align-middle"
            >
              <div
                className="time"
              >
                12:00
              </div>
              <div
                className="icon"
              >
                <img
                  alt="clear sky"
                  src="http://openweathermap.org/img/w/01d.png"
                />
              </div>
              <div
                className="temp"
              >
                286.164
                <sup>
                  o
                </sup>
                C
              </div>
              <div
                className="hpa"
              >
                1017.58 hPa
              </div>
              <div
                className="wind"
              >
                3.61
              </div>
              <div
                className="wind-arrow"
              >
                <i
                  className="material-icons"
                  style={
                    Object {
                      "transform": "rotate(165.001deg)",
                    }
                  }
                >
                  arrow_right_alt
                </i>
              </div>
            </li>
          </ul>
        </div>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5
              className="card-title"
            >
              Today (11 - Oct - 2018)
            </h5>,
            <ul
              className="list-group"
            >
              <li
                className="d-flex list-group-item align-middle"
              >
                <div
                  className="time"
                >
                  12:00
                </div>
                <div
                  className="icon"
                >
                  <img
                    alt="clear sky"
                    src="http://openweathermap.org/img/w/01d.png"
                  />
                </div>
                <div
                  className="temp"
                >
                  286.164
                  <sup>
                    o
                  </sup>
                  C
                </div>
                <div
                  className="hpa"
                >
                  1017.58 hPa
                </div>
                <div
                  className="wind"
                >
                  3.61
                </div>
                <div
                  className="wind-arrow"
                >
                  <i
                    className="material-icons"
                    style={
                      Object {
                        "transform": "rotate(165.001deg)",
                      }
                    }
                  >
                    arrow_right_alt
                  </i>
                </div>
              </li>
            </ul>,
          ],
          "className": "card-body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Today (11 - Oct - 2018)",
              "className": "card-title",
            },
            "ref": null,
            "rendered": "Today (11 - Oct - 2018)",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="d-flex list-group-item align-middle"
                >
                  <div
                    className="time"
                  >
                    12:00
                  </div>
                  <div
                    className="icon"
                  >
                    <img
                      alt="clear sky"
                      src="http://openweathermap.org/img/w/01d.png"
                    />
                  </div>
                  <div
                    className="temp"
                  >
                    286.164
                    <sup>
                      o
                    </sup>
                    C
                  </div>
                  <div
                    className="hpa"
                  >
                    1017.58 hPa
                  </div>
                  <div
                    className="wind"
                  >
                    3.61
                  </div>
                  <div
                    className="wind-arrow"
                  >
                    <i
                      className="material-icons"
                      style={
                        Object {
                          "transform": "rotate(165.001deg)",
                        }
                      }
                    >
                      arrow_right_alt
                    </i>
                  </div>
                </li>,
              ],
              "className": "list-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1539248400",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="time"
                    >
                      12:00
                    </div>,
                    <div
                      className="icon"
                    >
                      <img
                        alt="clear sky"
                        src="http://openweathermap.org/img/w/01d.png"
                      />
                    </div>,
                    <div
                      className="temp"
                    >
                      286.164
                      <sup>
                        o
                      </sup>
                      C
                    </div>,
                    <div
                      className="hpa"
                    >
                      1017.58 hPa
                    </div>,
                    <div
                      className="wind"
                    >
                      3.61
                    </div>,
                    <div
                      className="wind-arrow"
                    >
                      <i
                        className="material-icons"
                        style={
                          Object {
                            "transform": "rotate(165.001deg)",
                          }
                        }
                      >
                        arrow_right_alt
                      </i>
                    </div>,
                  ],
                  "className": "d-flex list-group-item align-middle",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "12:00",
                      "className": "time",
                    },
                    "ref": null,
                    "rendered": "12:00",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <img
                        alt="clear sky"
                        src="http://openweathermap.org/img/w/01d.png"
                      />,
                      "className": "icon",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "clear sky",
                        "src": "http://openweathermap.org/img/w/01d.png",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        286.164,
                        <sup>
                          o
                        </sup>,
                        "C",
                      ],
                      "className": "temp",
                    },
                    "ref": null,
                    "rendered": Array [
                      286.164,
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "o",
                        },
                        "ref": null,
                        "rendered": "o",
                        "type": "sup",
                      },
                      "C",
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "1017.58 hPa",
                      "className": "hpa",
                    },
                    "ref": null,
                    "rendered": "1017.58 hPa",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": 3.61,
                      "className": "wind",
                    },
                    "ref": null,
                    "rendered": 3.61,
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <i
                        className="material-icons"
                        style={
                          Object {
                            "transform": "rotate(165.001deg)",
                          }
                        }
                      >
                        arrow_right_alt
                      </i>,
                      "className": "wind-arrow",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "arrow_right_alt",
                        "className": "material-icons",
                        "style": Object {
                          "transform": "rotate(165.001deg)",
                        },
                      },
                      "ref": null,
                      "rendered": "arrow_right_alt",
                      "type": "i",
                    },
                    "type": "div",
                  },
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
